Derivs/src/derivs.hxx

```diff
@@ -1,519 +1,512 @@
+  if constexpr (deriv_order == 2)
+    return (load(1) - 2 * load0()) / pow2(dx);
+  if constexpr (deriv_order == 4)
+    return (1 / T(12) * load(2) - 4 / T(3) * load(1) + 5 / T(2) * load0()) /
+           pow2(dx);
+  if constexpr (deriv_order == 6)
+    return (1 / T(90) * load(3) - 3 / T(20) * load(2) + 3 / T(2) * load(1) -
+            49 / T(18) * load0()) /
+           pow2(dx);
+}
+
+using std::tuple_size_v;
+using Arith::min;
+using std::array;
+using Arith::div_floor;
+using Arith::div_ceil;
+using Arith::align_ceil;
+using Arith::mask_for_loop_tail;
+
+template <typename T>
+inline ARITH_INLINE ARITH_DEVICE ARITH_HOST simd<T>
+deriv2_2d(const int vavail, const simdl<T> &mask, const T *restrict const var,
+          const ptrdiff_t di, const ptrdiff_t dj, const T dx, const T dy) {
+  constexpr size_t vsize = tuple_size_v<simd<T> >;
+  if (di == 1) {
+    assert(vavail > 0);
+    constexpr int maxnpoints = deriv_order + 1 + vsize - 1;
+    const int npoints = deriv_order + 1 + min(int(vsize), vavail) - 1;
+    array<simd<T>, div_ceil(maxnpoints, int(vsize))> arrx;
+    for (int i = 0; i < maxnpoints; i += vsize) {
+      if (i < npoints) {
+        const simdl<T> mask1 = Arith::mask_for_loop_tail<simdl<T> >(i, npoints);
+        arrx[div_floor(i, int(vsize))] =
+            deriv1d(mask1, &var[i - deriv_order / 2], dj, dy);
       }
-      r /= divisor;
-      return std::move(r);
     }
-    if constexpr (antisymmetric) {
-      R r{0};
-      for (std::ptrdiff_t n = 0; n < N / 2; ++n) {
-        const std::ptrdiff_t n1 = N - 1 - n;
-        r += coeffs[n] * (arr(n + I0) - arr(n1 + I0));
+#ifdef CCTK_DEBUG
+    for (int i = npoints; i < align_ceil(maxnpoints, int(vsize)); ++i)
+      ((T *)&arrx[0])[i] = Arith::nan<T>()(); // unused
+#endif
+    const T *const varx = (T *)&arrx[0] + deriv_order / 2;
+    return deriv1d(mask, varx, 1, dx);
+  } else {
+    assert(dj != 1);
+    array<simd<T>, deriv_order + 1> arrx;
+    for (int j = -deriv_order / 2; j <= deriv_order / 2; ++j)
+      if (j == 0) {
+#ifdef CCTK_DEBUG
+        arrx[deriv_order / 2 + j] = Arith::nan<simd<T> >()(); // unused
+#endif
+      } else {
+        arrx[deriv_order / 2 + j] = deriv1d(mask, &var[j * dj], di, dx);
       }
-      r /= divisor;
-      return std::move(r);
-    }
-    R r{0};
-    for (std::ptrdiff_t n = 0; n < N; ++n)
-      r += coeffs[n] * arr(n + I0);
-    return std::move(r);
+    const T *const varx = (T *)(&arrx[deriv_order / 2]);
+    return deriv1d(mask, varx, vsize, dy);
   }
-};
-
-// Interpolate at i = 0
-constexpr stencil<0, 0, symmetric> interp{1, {1}};
-
-// Derivative at i = 0
-constexpr stencil<-1, +1, antisymmetric> deriv1_o2{2, {-1, 0, +1}};
-
-constexpr stencil<-2, +2, antisymmetric> deriv1_o4{12, {-1, +8, 0, -8, +1}};
-
-constexpr stencil<-1, +1, symmetric> deriv2_o2{1, {-2, +1, -2}};
-
-constexpr stencil<-2, +2, symmetric> deriv2_o4{12, {-1, +16, -30, +16, -1}};
-
-// Interpolate at i = 1/2
```

Comment:
These stencils are new capabilities in thorn `Derivs` that are not present in the thorns `Z4c` and `Weyl`. If you move the `Z4c` and `Weyl` code over, and delete the existing code in `Derivs`, then you are removing capabilities. The current implementation of thorn `Derivs` is a more "modern" design than those in `Z4c` or `Weyl`, which kind-of grew wild.